app/src/main/java/openfoodfacts/github/scrachx/openfood/views/TipBox.java

```diff
@@ -59,21 +57,17 @@ public TipBox(Context context, @Nullable AttributeSet attrs) throws Exception {
         });
 
         prefs = PreferenceManager.getDefaultSharedPreferences(getContext());
-        boolean show = prefs.getBoolean(identifier, true);
-        if (!show) {
-            return;
+        //Tooltip should be shown the first time the actual content is available
+        boolean canDisplayImmediately = attributes.getBoolean(R.styleable.TipBox_shouldDisplayImmediately, false);
```

Comment:
I would prefer if we got the attributes that we need immediately so we can call `recycle()` as fast as we can. 

Please move `canDisplayImmediately` near the others `attributes.get...` and create a var for `attributes.getColor` s.

Reply:
Done